EventListener/LogoutEventListener.php

```diff
@@ -43,55 +42,51 @@ public function __construct(
             'secure' => true,
             'remove_token_from_body' => true,
         ], $cookieSettings);
-        $this->logout_firewall_context = $logout_firewall_context;
     }
 
     public function onLogout(LogoutEvent $event): void
     {
         $request = $event->getRequest();
-        $current_firewall_context = $request->attributes->get('_firewall_context');
-
-        if ($current_firewall_context !== $this->logout_firewall_context) {
-            return;
-        }
 
         $tokenString = $this->refreshTokenExtractor->getRefreshToken($request, $this->tokenParameterName);
+
         if (null === $tokenString) {
             $event->setResponse(
                 new JsonResponse(
                     [
                         'code' => 400,
                         'message' => 'No refresh_token found.',
                     ],
-                    JsonResponse::HTTP_BAD_REQUEST
+                    Response::HTTP_BAD_REQUEST
```

Comment:
BTW, we should not overload a new response if one exists and let the "DefaultLogoutListener" set it
And also set a negative priority to be before the "CookieClearingLogoutListener"

Reply:
> BTW, we should not overload a new response if one exists and let the "DefaultLogoutListener" set it

Well, this is a bit of a chicken vs. egg problem.  The `DefaultLogoutListener::onLogout()` listener has a priority of 64, which makes it the first event listener to run on the logout event in the app for one of my company's bigger clients.  So an `if ($event->getResponse() === null)` check isn't going to have a lot of value here without making it a higher priority than the default listener.

> And also set a negative priority to be before the "CookieClearingLogoutListener"

The `CookieClearingLogoutListener::onLogout()` listener already has a priority of -255, so this listener is already running before that one (when it's enabled).

Reply:
Tell me if I'm wrong: in case your application uses a form_login on the same firewall as the refresh_jwt, the `{firewall}.logout.path` configuration will not work anymore because the response created by the `DefaultLogoutListener` will be overwritten by this listener.
And BTW, even in the current configuration of this bundle, we should have this case

Reply:
In that scenario, yeah, you're hosed.

This almost needs a `generate_response` config param TBH, one can then explicitly opt into whether this bundle generates (and potentially overwrites a previously set) response for the logout request.

You then need to split the cookie invalidation part of the listener off into another lower priority listener (maybe the same priority as `CookieClearingLogoutListener::onLogout()`) that way it runs as late in the process as possible and has a better chance of clearing the cookie on the right response object (in case another bundle or a listener in the app itself is doing something with the response).

Reply:
Yeah a config on the refresh_jwt like `logout_path` to check if the route match to generate the json response or not